iml-agent/src/action_plugins/lamigo.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::agent_error::ImlAgentError;
+use futures::future::TryFutureExt;
+use std::path::Path;
+use std::process::Output;
+use std::{fmt, io};
+use tokio::fs;
+use tokio::process::{Child, Command};
+
+static SYSTEMD_DIR: &str = "/Users/nick/tmp/system";
+
+#[derive(serde::Deserialize, structopt::StructOpt, Clone, Debug)]
+pub struct Config {
+    #[structopt(long)]
+    /// File system name
+    fs: String,
+
+    #[structopt(long)]
+    /// MDT device that provides changelogs
+    mdt: String,
+
+    #[structopt(long)]
+    /// The user on which behalf lamigo consumes changelogs from the MDT device
+    user: String,
+
+    #[structopt(long)]
+    /// Fast OST pool name
+    hot_pool: String,
+
+    #[structopt(long)]
+    /// Slow OST pool name
+    cold_pool: String,
+
+    #[structopt(long)]
+    /// Interval for lamigo to wait before replicating a closed file (in seconds)
+    min_age: u32,
+
+    #[structopt(long)]
+    /// Lustre client mount point, e.g. `/mnt/lustre`
+    mount_point: String,
+
+    #[structopt(long)]
+    /// Lustre device to be mounted, e.g. `192.168.0.100@tcp0:/spfs`
+    lustre_device: String,
+
+    #[structopt(long)]
+    /// (Optional argument)
+    /// By default, nothing is performed, if the mount point is already mounted.
+    /// This flag is to force the mount execution.
+    force: bool,
+}
+
+impl fmt::Display for Config {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "[Unit]\n\
+             Description=Run lamigo service\n\
+             [Service]\n\
+             ExecStart=/usr/bin/lamigo \
+             -m {fs}-{mdt} \
+             -u {user} \
+             -s {hot_pool} \
+             -t {cold_pool} \
+             -a {min_age} \
+             {mount_point}\n\
+             ",
+            fs = self.fs,
+            mdt = self.mdt,
+            cold_pool = self.cold_pool,
+            hot_pool = self.hot_pool,
+            min_age = self.min_age,
+            user = self.user,
+            mount_point = self.mount_point,
+        )
+    }
+}
+
+pub async fn create_lamigo_service_unit(c: Config) -> Result<(), ImlAgentError> {
+    if c.force {
+        create_lamigo_service_unit_internal(SYSTEMD_DIR, &c)
+            .err_into()
+            .await
+    } else {
+        let is_mounted = is_filesystem_mounted(&c.mount_point).await?;
+        if !is_mounted {
+            mount_filesystem(&c).await?;
+        }
+        create_lamigo_service_unit_internal(SYSTEMD_DIR, &c).await?;
```

Comment:
This line looks duplicating `create_lamigo_service_unit_internal(SYSTEMD_DIR, &c)` above. Could it be something like this:

```rust
if !c.force {
  // mount stuff here
}
create_lamigo_service_unit_internal(SYSTEMD_DIR, &c).await?;
```

Reply:
Yes, I tried to explicitly state that "here the things we do when force is true, here is otherwise". I'm ok with your proposal.